pkg/actuators/machine/actuator.go

```diff
@@ -134,7 +135,11 @@ func (a *Actuator) updateMachineStatus(machine *machinev1.Machine, awsStatus *pr
 	}
 
 	oldAWSStatus := &providerconfigv1.AWSMachineProviderStatus{}
-	if err := a.codec.DecodeProviderStatus(machine.Status.ProviderStatus, oldAWSStatus); err != nil {
+	//if err := a.codec.DecodeProviderStatus(machine.Status.ProviderStatus, oldAWSStatus); err != nil {
```

Comment:
Can you add to do here explaining why is needed?

Reply:
Instead of replacing `a.codec.DecodeProviderStatus` with `yaml.Unmarshal`, let's replace the code in `DecodeProviderStatus` itself [1].

[1] https://github.com/openshift/cluster-api-provider-aws/blob/master/pkg/apis/awsproviderconfig/v1alpha1/register.go#L110